tests/components/filesize/test_sensor.py

```diff
@@ -21,40 +21,36 @@ def create_file(path):
         test_file.write("test")
 
 
-class TestFileSensor(unittest.TestCase):
-    """Test the filesize sensor."""
+@pytest.fixture(autouse=True)
+def remove_file():
+    """Remove test file."""
+    if os.path.isfile(TEST_FILE):
+        os.remove(TEST_FILE)
 
-    def setup_method(self, method):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.hass.config.allowlist_external_dirs = {TEST_DIR}
 
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        if os.path.isfile(TEST_FILE):
-            os.remove(TEST_FILE)
-        self.hass.stop()
+async def test_invalid_path(hass):
+    """Test that an invalid path is caught."""
+    config = {"sensor": {"platform": "filesize", CONF_FILE_PATHS: ["invalid_path"]}}
+    assert await async_setup_component(hass, "sensor", config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)